testhelpers/steps.go

```diff
@@ -566,3 +592,41 @@ func (ctx *TestContext) TableAtIDShouldBe(tableName string, id int64, data *gher
 	}
 	return nil
 }
+
+func (ctx *TestContext) TableHasUniqueKey(tableName, indexName, columns string) error { // nolint
+	db, err := gorm.Open("mysql", ctx.db())
+	if err != nil {
+		return err
+	}
+	defer func() { _ = db.Close() }() // nolint: gosec
+
+	if db.Dialect().HasIndex(tableName, indexName) {
+		return nil
+	}
+
+	if err = db.Table(tableName).AddUniqueIndex(indexName, strings.Split(columns, ",")...).Error; err != nil {
+		return err
+	}
+	ctx.addedDBIndices = append(ctx.addedDBIndices, &addedDBIndex{Table: tableName, Index: indexName})
+	return nil
+}
+
+func (ctx *TestContext) TheGeneratedGroupPasswordIs(generatedPassword string) error { // nolint
+	monkey.Patch(groups.GenerateGroupPassword, func() (string, error) { return generatedPassword, nil })
+	return nil
+}
+
+var multipleStringsRegexp = regexp.MustCompile(`^"([^"]*)"(?:\s*,\s*"([^"]*)")*$`)
+
+func (ctx *TestContext) TheGeneratedGroupPasswordsAre(generatedPasswords string) error { // nolint
+	currentIndex := 0
+	passwords := multipleStringsRegexp.FindAllStringSubmatch(generatedPasswords, -1)
+	monkey.Patch(groups.GenerateGroupPassword, func() (string, error) {
+		currentIndex++
+		if currentIndex >= len(generatedPasswords) {
+			return "", errors.New("not enough generated passwords")
+		}
+		return passwords[0][currentIndex], nil
+	})
+	return nil
+}
```

Comment:
The next time we add something in this file, we should split it. It becomes too long and messy.

Reply:
kk